hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2218,3 +2218,77 @@ def hpat_pandas_series_median_impl(self, axis=None, skipna=True, level=None, num
+
+       Tests: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_axis1
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_axis2
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_axis3
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_float_index1
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_float_index2
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_str_index1
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_str_index2
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_str_index3
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_float_inplace_no_index1
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_float_inplace_no_index2
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_str_inplace_no_index1
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_str_inplace_no_index2
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_str_parallel1
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_dt_no_index1
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_bool_no_index1
+              python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_dropna_int_no_index1
+
+    Parameters
+    ----------
+    self: :obj:`pandas.Series`
+        input series
+    axis: :obj:`int` or :obj:`string` {0 or `index`}, default 0
+        There is only one axis to drop values from.
+    inplace: :obj:`bool`, default False
+        If True, do operation inplace and return None.
+        *unsupported*
+
+    Returns
+    -------
+    :obj:`pandas.Series`
+         returns :obj:`pandas.Series` object with NA entries dropped from it.
+    """
+
+    _func_name = 'Method dropna().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+
+    if not (isinstance(axis, (types.Integer, types.StringLiteral, types.UnicodeType, types.Omitted)) or axis == 0):
+        raise TypingError('{} The axis must be an Integer or String. Given: {}'.format(_func_name, axis))
+
+    if not (inplace is False or isinstance(inplace, types.Omitted)):
+        raise TypingError('{} Unsupported parameters. Given inplace: {}'.format(_func_name, inplace))
+
+    # TODO: handle inplace argument (currently supported with old-style impl only via assign.target)
+    # Find a way to assign back to self._data and return None
+    if isinstance(self.dtype, (types.Number, types.Boolean)):
```

Comment:
Are we able to merge if and else statements? I mean to use one of the approaches for both cases.

Reply:
@densmirn Yes, the else branch actually handles both cases. But as in other places legacy implementation tends to provide lightweight overloads where possible. I will add new gen_nan_mask function for arrays in the next commit.